tests/ui/traits/const-traits/staged-api.rs

```diff
@@ -37,8 +37,7 @@ const fn const_context() {
     //[stable]~^^ ERROR not yet stable as a const fn
     Foo::func();
     //[unstable]~^ ERROR cannot use `#[feature(local_feature)]`
-    //[stable]~^^ cannot be (indirectly) exposed to stable
-    // We get the error on `stable` since this is a trait function.
+    // FIXME: in [stable] we should still be complaining about this being a trait call.
```

Comment:
Trait calls don't show any errors any more... but the old trait call checks are broken anyway so I figured it's fine to break them even more, and wait until @compiler-errors lands the new checks. :)

Reply:
See [here](https://github.com/rust-lang/rust/pull/132479#issuecomment-2454641534) for how I think this check should work.

I can add that to this PR, if you prefer.

Reply:
After rebasing this PR no longer affects the const-trait tests. :)